OverlayManager refactor and Flash

rneuser opened the pull request:
* Refactors OverlayManager to be more versatile. You can now add multiple overlays to clients, as well as send parameters along for a specific overlay if desired.

* Adds the flash item from SS13

* MeleeWeaponComponent.Attack now returns when OnHitEntities returns false

ShadowCommander commented on Resources/Textures/Objects/Melee/flash.rsi/meta.json:
```diff
@@ -0,0 +1,76 @@
+        "x": 32,
+        "y": 32
+    },
+    "license": "GNU AGPL v3",
```
This is a code license, you'll want the find the content license which for SS13 is usually CC BY-SA of some sort.

rneuser commented:
Well so I changed it to a draft PR because I'm not sure this is really the best way to go about this so lemme know. Some code is definitely in non-prime locations, but that's just a minor issue. 

Essentially now instead of the OverlayManager being able to only set one overlay at a time, I changed it to be able to handle multiple overlays that can have their own state that is sent to the client.

Big thing that's probably not ideal is the IConfigurable interface, which is implemented on the actual Overlay class to be able to apply state changes. Might be a bit too generic? idk

ShadowCommander commented on Content.Server/GameObjects/Components/Weapon/Melee/FlashComponent.cs:
```diff
@@ -0,0 +1,108 @@
+            return true;
+        }
+
+        public bool UseEntity(UseEntityEventArgs eventArgs)
```
Make sure to remove this

rneuser replied:
This is now AoE like the ss13 flash

ShadowCommander commented on Content.Shared/GameObjects/Components/Mobs/SharedOverlayEffectsComponent.cs:
```diff
@@ -13,21 +21,70 @@ public abstract class SharedOverlayEffectsComponent : Component
-        CircleMask,
-        GradientCircleMask,
+        [ViewVariables(VVAccess.ReadOnly)]
+        public string ID { get; }
```
I'm not sure it matters, but you might be able to use an enum here instead.

rneuser replied:
I think I will add an enum to Content.Shared so the server and client can agree on overlay ID's properly. Caveat being when someone adds an overlay they have to also add an entry to the enum. I'll probably add a test or something to check for that.
